### PR TITLE
Throw error when trying to start on Wayland

### LINKED ISSUE #243: i3lock can be killed on wayland
## I'm submitting a…

<pre>
[ ] Bug
[ ] Feature Request
[X] Other: Vulnerability
</pre>

## Current Behavior
If i3lock is running on xwayland shortcuts still work. This especially include positioning some window in front of i3lock. Just search a terminal and call killall i3lock – or whatever you want.

## Expected Behavior
i3lock should lock your screen and should not be able to be disabled.

## Reproduction Instructions
1. add bindsym $mod+m exec killall i3lock to your sway config
2. start i3lock 
3. press $mod+m

## Proposed workaround
show a warning message instead of locking the screen, if you detect running on a wayland compositor.

## Environment
Output of `i3lock --version`:
<pre>
i3lock version: 2.11.1
</pre>

### PR DESCRIPTION
**Pull request by v0idpwn**
As i3lock doesn't support Wayland and the wayland userbase is growing day after day, I think this fits pretty well.

Closes #243

### CODE REVIEW

File: i3lock.c [hunk 1]
```diff
@@ -1038,6 +1038,8 @@ int main(int argc, char *argv[]) {
         err(EXIT_FAILURE, "getpwuid() failed");
     if ((username = pw->pw_name) == NULL)
         errx(EXIT_FAILURE, "pw->pw_name is NULL.");
+    if getenv("WAYLAND_DISPLAY")
```
Comment (stapelberg) [inline i3lock.c:1041]: I don’t think this compiles? I thought parens are required in C.

Please make this read:

```C
if (getenv("WAYLAND_DISPLAY") != NULL) {
  errx(EXIT_FAILURE, "i3lock is a program for X11 and does not work on Wayland. Try https://github.com/swaywm/swaylock instead");
}
```

**Review by stapelberg — CHANGES_REQUESTED** [review]

File: i3lock.c (hunk 1 above)
Comment (v0idpwn) [reply i3lock.c:1041]: You're right. I was just ammending, long time I don't do C.

**Comment by v0idpwn** [comment]
Just did the requested changes

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Thanks!

**Pull request merged**